monitoring/README.md

```diff
@@ -1,112 +1,87 @@
-```
+This example shows how to deploy monitoring tools (Prometheus, Grafana and Pyroscope) for KubeStellar control plane components - see architecture image above. These instructions can be used to deploy monitoring tools to any KubeStellar system. If you already have the monitoring tools installed in your environment, you can skip to `step-2`.
+
+1. Install the monitoring tools:
+
+    Starting from a local directory containing the git repo, do the following:
+
+    ```bash
+    $ cd monitoring
+    $ ./install-ks-monitoring.sh
+    ```
+
+    Optionally, check the deployment of the monitoring tools:
+
+    ```bash
+    $ kubectl -n ks-monitoring get pods
+    NAME                                                  READY   STATUS    RESTARTS   AGE
+    grafana-7969f6cf5c-lp4g6                              1/1     Running   0          66s
+    prometheus-kube-prometheus-operator-b56d747c9-wfs2w   1/1     Running   0          42s
+    prometheus-prometheus-kube-prometheus-prometheus-0    2/2     Running   0          39s
+    pyroscope-0                                           1/1     Running   0          90s
+    pyroscope-alloy-0                                     2/2     Running   0          90s
+    ```
+
+2. Configure Prometheus and Pyroscope to scrape your KubeStellar spaces:
+
+    ```bash
+    $ ./configure-metrics-tools.sh
+    ```
+
+    Optionally, check the prometheus service monitor objects:
+
+    ```bash
+    $ kubectl -n ks-monitoring get servicemonitor -l "app.kubernetes.io/part-of!=kube-prometheus-stack"
+    NAME                      AGE
+    ks-controller-manager     70m
+    ks-transport-controller   70m
+    wds-apiserver             70m
+    ```
+
+
+3. Connect to Grafana UI: 
+
+    a) Port-forward Grafana to localhost, by using the kubectl command:
+
+    ```bash
+    $ kubectl port-forward -n ks-monitoring svc/grafana 3000:80
+    ```
+
+    b) Check Prometheus and Pyroscope data source configuration: in a browser, go to the Grafana server at http://localhost:3000 and on the left-hand side, go to Home > Connections > Data sources.
+
+    <img src="images/grafana-data-sources.png" width="60%" height="60%" title="grafana-data-sources">
+
+
+4. Import KubeStellar Grafana dashboards into the Grafana UI:
+
+    a) Click `Dashboards` in the primary menu
+
+    b) Click `New` and select `Import` in the drop-down menu
+
+    c) Upload the dashboard JSON files from your local `./grafana` sub-directory into the Grafana UI. You should be able to see the following dashboards:
+
+    <img src="images/grafana-dashboard-view.png" width="60%" height="60%" title="grafana-dashboard-view">
+
+
+5. View Grafana dashboard based on KubeStellar WDS space and controller component names:
+
+    a) WDS API Server: input into the following text boxes.
+    - `SpaceNS`: name of the KubeStellar space namespace, e.g., *wds1-system*, *wds2-system*, etc.
+    - `SpaceName`: name of the KubeStellar space, e.g., *wds1*, *wds2*, etc.
```

Comment:
What are the combinations that get me data? After `test/e2e/run-test.sh` I find only two:
- SpaceNS=wds1-system, SpaceName=wds1
- SpaceNS=default, SpaceName=apiserver (what is this one about?)

<img width="896" alt="Screenshot 2024-08-13 at 4 35 44 PM" src="https://github.com/user-attachments/assets/0904b74f-0578-4ee9-9f1f-f848de825b9d">

Reply:
@MikeSpreitzer, for the moment I am not displaying the data from the local apiserver yet. I will add that feature into another NEW PR. I want to first have a template of graphs in the dashboard that we all agree before making further changes.

Reply:
FYI, I am already scraping with prometheus the hosting cluster apiserver too.

Reply:
(A) That is worth documenting.
(B) What about the its1 apiserver?

Reply:
I will push a new PR with update documentation, including fixing the issue with scraping the its1 space.